shared/ci/tasks/cache-pnpm-deps.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+
+set -eu
+
+tar_out="$(pwd)/bundled-deps"
+
+pushd deps
+# Install dependencies
+echo "  * pnpm install"
+pnpm install --shamefully-hoist
```

Comment:
why is this required? are you getting errors? `--shamefully-hoist`

Reply:
Yes. That was some bigger headache yesterday. pnpm uses caching mechanisms by hardlinks. Those hardlinks need then additional considerations regarding the (un-)tar-ing. 
As we explicitely use a tar-file for our dependencies, there is no way and also no need to use sophisticated caching mechanisms. So ideally, pnpm behaves exactly like npm (or yarn) but that does not seem to be the case exactly. 

Anyway, this is as close as we can get and it works.